### PR TITLE
Add script for test coverage pipeline 

### LINKED ISSUE #565: Add coverage pipeline
### What would you like to be added?

Currently, we don't have pipeline to calculate test coverage for our unit test.

### Why is this needed?

To improve code quality

### Example
https://github.com/kubernetes/test-infra/blob/master/config/jobs/kubernetes-sigs/cluster-api-provider-aws/cluster-api-provider-aws-periodics.yaml#L179-L211

### CPI's pipeline file
https://github.com/kubernetes/test-infra/blob/master/config/jobs/kubernetes/cloud-provider-vsphere/cloud-provider-vsphere-config.yaml

### PR DESCRIPTION
**Pull request by chenlin07**
Signed-off-by: Chen Lin <linch@vmware.com>

**What this PR does / why we need it**:
We want a pipeline to calculate test coverage for unit tests

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #565

**Special notes for your reviewer**:

**Release note**:

```release-note
```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -229,9 +229,11 @@ build-unit-tests:
 test: unit
 build-tests: build-unit-tests
 
-.PHONY: cover
-cover: TEST_FLAGS += -cover
-cover: test
+.PHONY: test-cover
+test-cover:  ## Run tests with code coverage and code generate reports
+	go test -v -coverprofile=coverage.out ./...
```
Comment (chenlin07) [inline Makefile:234]: we need coverage.out file for test coverage pipeline in test-infra

Comment (HanFa) [reply Makefile:234]: if this file is never needed in VCS, shall we add it into `.gitignore`?

Comment (chenlin07) [reply Makefile:234]: Good catch! Added into `.gitignore`

Comment (lubronzhan) [reply Makefile:234]: Also need
```
*.out
coverage.*
```

Also the coverage output

Comment (chenlin07) [reply Makefile:234]: fixed! thanks

**Pull request merged**